community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -108,9 +110,20 @@
     /**
      * Closes this transaction, committing its changes iff {@link #success()} has been called and
      * {@link #failure()} has NOT been called. Otherwise its changes will be rolled back.
+     *
+     * @return id of the committed transaction or {@link #NOT_COMMITTED} if transaction was rolled back or read-only.
```

Comment:
Sure but that is not what the doc says.
"Closes this transaction, committing its changes if and only if `success()` has been called and `failure()` has NOT been called. Otherwise its changes will be rolled back."
But if `success()` has been called and `failure()` has not, the transaction will still not terminate if `markForTermination()` has been called.